### PR TITLE
Make getBlocksFrom() in Ledger return a range

### PR DESCRIPTION
**Pull request by AndrejMitrovic**
This makes it easier to optimize & cache the result at the call site (which is not done yet, but it enables it).

### CODE REVIEW

File: source/agora/node/Ledger.d [hunk 1]
```diff
@@ -283,37 +284,30 @@ public class Ledger
...
         Returns:
-            the array of blocks starting from block_height,
-            up to `max_blocks`
+            the range of blocks starting from start_height
 
     ***************************************************************************/
 
-    public const(Block)[] getBlocksFrom (ulong block_height, size_t max_blocks)
-        @safe
+    public auto getBlocksFrom ( size_t start_height ) @safe nothrow
```
Comment (MichaelKim20) [inline source/agora/node/Ledger.d]:
```suggestion
    public auto getBlocksFrom (size_t start_height) @safe nothrow
```

**Review by MichaelKim20 — COMMENTED** [review]
This PR is expected to have a lot of effect on performance.

**Comment by AndrejMitrovic** [comment]
Fixed styling issue.

> This PR is expected to have a lot of effect on performance.

Not yet, but it will enable caching in the future. I will need this code for SCP though, to restore the SCP state when a validator node boots up.

**Review by MichaelKim20 — APPROVED** [review]

**Review by Geod24 — APPROVED** [review]
Neat

**Pull request merged**